packages/react-sdk/README.md

```diff
@@ -167,6 +167,142 @@ const {
+
+### Using `ReflagBootstrappedProvider`
+
+The `ReflagBootstrappedProvider` is designed for server-side rendering (SSR) scenarios where you've pre-fetched flags on the server and want to pass them to the client without an additional network request.
+
+```tsx
+import { useState, useEffect } from "react";
+import { BootstrappedFlags } from "@reflag/react-sdk";
+
+interface BootstrapData {
+  user: User;
+  flags: BootstrappedFlags;
+}
+
+function useBootstrap() {
+  const [data, setData] = useState<BootstrapData | null>(null);
+
+  useEffect(() => {
+    fetch("/bootstrap")
+      .then((res) => res.json())
+      .then(setData);
+  }, []);
+
+  return data;
+}
+
+// Usage in your app
+function App() {
+  const { user, flags } = useBootstrap();
+
+  return (
+    <AuthProvider user={user}>
+      <ReflagBootstrappedProvider
+        publishableKey="your-publishable-key"
+        flags={flags}
+      >
+        <Router />
+      </ReflagBootstrappedProvider>
+    </AuthProvider>
+  );
+}
+```
+
+### Server-side endpoint setup
+
+Create an endpoint that provides bootstrap data to your client application:
+
+```typescript
+// server.js or your Express app
+import { ReflagClient as ReflagNodeClient } from "@reflag/node-sdk";
+
+const reflagClient = new ReflagNodeClient({
+  secretKey: process.env.REFLAG_SECRET_KEY,
+});
+await reflagClient.initialize();
+
+app.get("/bootstrap", (req, res) => {
+  const user = getUser(req); // Get user from your auth system
+  const company = getCompany(req); // Get company from your auth system
+
+  const flags = reflagClient.getFlagsForBootstrap({
+    user: { id: user.id, email: user.email, role: user.role },
+    company: { id: company.id, plan: company.plan },
+    other: { source: "web" },
+  });
+
+  res.status(200).json({
+    user,
+    flags,
+  });
+});
+```
+
+### Next.js SSR example
+
+For Next.js applications using server-side rendering, you can pre-fetch flags in `getServerSideProps`:
+
+```typescript
+// pages/index.tsx
+import { GetServerSideProps } from "next";
+import { ReflagClient as ReflagNodeClient } from "@reflag/node-sdk";
+import { ReflagBootstrappedProvider, BootstrappedFlags, useFlag } from "@reflag/react-sdk";
+
+interface PageProps {
+  bootstrapData: BootstrappedFlags;
+}
+
+export const getServerSideProps: GetServerSideProps = async (context) => {
+  const serverClient = new ReflagNodeClient({
+    secretKey: process.env.REFLAG_SECRET_KEY
+  });
+  await serverClient.initialize();
+
+  const user = await getUserFromSession(context.req);
+  const company = await getCompanyFromUser(user);
+
+  const bootstrapData = serverClient.getFlagsForBootstrap({
+    user: { id: user.id, email: user.email, role: user.role },
+    company: { id: company.id, plan: company.plan },
+    other: { page: "homepage" }
+  });
+
+  return { props: { bootstrapData } };
+};
+
+export default function HomePage({ bootstrapData }: PageProps) {
+  return (
+    <ReflagBootstrappedProvider
```

Comment:
awesome to get a Next.js example in there! good work. Have you tried it out end to end? I read somewhere that context providers only work in the client .. or something..

Reply:
And now a full demo that works (if configured) https://github.com/reflagcom/javascript/pull/475/commits/e283941ea26691eb150f57883dbf109edac7b0cb